docs/developer-guide/plugins-scenarios.md

```diff
@@ -233,115 +231,79 @@ However there are some pitfalls you need to know:
     @Override
-    public StyxObservable<HttpResponse> intercept(HttpRequest request, Chain chain) {
-        StyxObservable<ByteBuf> toUpperCase = request.body()
-                .map(buf -> {
-                    buf.release();
-                    return copiedBuffer(buf.toString(UTF_8).toUpperCase(), UTF_8);
-                });
-
+    public Eventual<LiveHttpResponse> intercept(LiveHttpRequest request, Chain chain) {
         return chain.proceed(
                 request.newBuilder()
-                        .body(toUpperCase)
+                        .body(byteStream ->                                                              // 1
+                                byteStream.map(buf -> {                                                  // 2
+                                    String upperCase = new String(buf.content(), UTF_8).toUpperCase();   // 3
+                                    return new Buffer(upperCase, UTF_8);
+                                }))
                         .build()
         );
     }
 }    
 ```
 
-### Asynchronously transforming streaming request content
-
-Asynchronous content stream transformation is very similar to the synchronous transformation.
-The only difference is that asynchronous transformation must be composed with a `flatMap` 
-instead of `map`. Otherwise the same discussion apply. As always it is important to take care
-of the reference counting.
+A call to `request.newBuilder` opens up a new request builder that allows request to be transformed.
 
-```java
-public class AsyncRequestContentTransformation implements Plugin {
-
-    @Override
-    public StyxObservable<HttpResponse> intercept(HttpRequest request, Chain chain) {
-        StyxObservable<ByteBuf> contentMapping = request.body()
-                .flatMap(buf -> {
-                    String content = buf.toString(UTF_8);
-                    buf.release();
-                    return sendToRemoteServer(content)
-                            .map(value -> copiedBuffer(value, UTF_8));
-                });
+The body transformation involves two lambda methods:
 
-        return chain.proceed(
-                request.newBuilder()
-                        .body(contentMapping)
-                        .build()
-        );
-    }
-    
-    StyxObservable<String> sendToRemoteServer(String buf) {
-        return StyxObservable.of("modified 3rd party content");
-    }
-}    
-```
+  1. A `.body(Function<ByteStream, ByteStream>)` accepts a lambda that modifies the request
+     byte stream. Here we provide a lambda that accepts a `ByteStream` and returns another
+     by applying a synchronous `map` operator on the stream.
+     
+  2. The `ByteStream` is modified by applying a `map` operator that synchronously modifies
+     each `Buffer` that streams through. The `map` involves another lambda that accepts a
+     `Buffer` and returns a modified buffer.
+  
+  3. In this example the buffer is converted to an upper case. 
```

Comment:
"In this example the buffer is converted to an upper case. " -> "In this example, the content of the buffer is interpreted as a String and converted to upper case. "

Reply:
👍